packages/fuels-core/src/json_abi.rs

```diff
@@ -1848,4 +1855,264 @@ mod tests {
+        );
+
+        let value = "Hd57a9c46dfcc7f18207013e65b44e4cb4e2c2298f4ac457ba8f82743f31e90b".to_string();
+        let error_message = abi
+            .tokenize(&ParamType::B256, value)
+            .unwrap_err()
+            .to_string();
+
+        assert!(error_message.contains("Parse hex error: Invalid character"));
+    }
+
+    #[test]
+    fn tokenize_errors_tuple() {
+        let params = Property {
+            name: "input".to_string(),
+            type_field: "(u64, [u64; 3])".to_string(),
+            components: Some(vec![
+                Property {
+                    name: "__tuple_element".to_string(),
+                    type_field: "u64".to_string(),
+                    components: None,
+                },
+                Property {
+                    name: "__tuple_element".to_string(),
+                    type_field: "[u64; 3]".to_string(),
+                    components: None,
+                },
+            ]),
+        };
+
+        let abi = ABIParser::new();
+
+        if let ParamType::Tuple(tuple_params) = parse_tuple_param(&params).unwrap() {
+            let error_message = abi
+                .tokenize_tuple("0, [0,0,0])", &tuple_params)
+                .unwrap_err()
+                .to_string();
+
+            assert_eq!(
+                "Invalid data: tuple value string must start and end with round brackets",
+                error_message
+            );
+
+            let error_message = abi
+                .tokenize_tuple("((0, [0,0,0])", &tuple_params)
+                .unwrap_err()
+                .to_string();
+
+            assert_eq!(
+                "Invalid data: tuple value string has excess opening brackets",
+                error_message
+            );
+
+            let error_message = abi
+                .tokenize_tuple("(0, [0,0,0]))", &tuple_params)
+                .unwrap_err()
+                .to_string();
+
+            assert_eq!(
+                "Invalid data: tuple value string has excess closing brackets",
+                error_message
+            );
+
+            let error_message = abi
+                .tokenize_tuple("(0, \"[0,0,0])", &tuple_params)
+                .unwrap_err()
+                .to_string();
+
+            assert_eq!(
+                "Invalid data: tuple value string has excess quotes",
+                error_message
+            );
+
+            let error_message = abi
+                .tokenize_tuple("(0, [0,0,0], 0, 0)", &tuple_params)
+                .unwrap_err()
+                .to_string();
+
+            assert_eq!(
+                "Invalid data: tuple value is missing a matching param",
+                error_message
+            );
+
+            let error_message = abi
+                .tokenize_tuple("(0, [0,0,0], 0)", &tuple_params)
+                .unwrap_err()
+                .to_string();
+
+            assert_eq!(
+                "Invalid data: last tuple value is missing a matching param",
```

Comment:
`has more elements than declared parameter types` or something would make more sense, wdyt?

Reply:
I agree that we should change it. However, I would not use the word _elements_ as it could be even more confusing. My proposal: we remove the word _last_ and use just `"Invalid data: tuple value is missing a matching parameter type"`

Reply:
`missing a matching parameter` doesn't convey that basically the user has more parameter types than value, "matching" is too vague. `tuple value contains more elements than the parameter types provided` is good I think.